src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -90,12 +90,37 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
             return;
         }
 
+        self::inline($classes, $cache, $declared);
+
+        if ($autoReload) {
+            // save the resources
+            self::writeCacheFile($metadata, serialize(array($files, $classes)));
```

Comment:
All use cases I saw so far need to write the cache. The metadata-related logic looks fine to me as a decorator thing, handled by `::load()`.